app/src/main/java/com/duckduckgo/app/browser/duckplayer/DuckPlayerJSHelper.kt

```diff
@@ -239,6 +240,33 @@ class DuckPlayerJSHelper @Inject constructor(
             "reportPageException", "reportInitException" -> {
                 Timber.tag(method).d(data.toString())
             }
+            "telemetryEvent" -> {
+                val attributes = data?.getJSONObject("attributes")
+
+                if (attributes?.getString("name") == "impression" && attributes.getString("value") == "landscape-layout") {
+                    pixel.fire(DUCK_PLAYER_LANDSCAPE_LAYOUT_IMPRESSIONS)
+                }
+
+                // TODO (cbarreiro) Abstract this to provide better support for telemetry events
+                /**
+                 * incoming data looks like this, where `name` is used to discriminate,
+                 * and 'value' is linked to it (but is optional)
+                 *
+                 * {
+                 *   "attributes": {
+                 *     "name": "impression",
+                 *     "value": "landscape-layout"
+                 *   }
+                 * }
+                 *
+                 * Another event might look like this in the future: (note: no 'value' field)
+                 * {
+                 *   "attributes": {
+                 *     "name": "page-view"
+                 *   }
+                 * }
+                 */
+            }
```

Comment:
Is there a follow-up task for this?

Reply:
I wasn't planning to create one for now, but to keep the TODO as a reminder in case we get new events. I'm open to discussing the possibility to create a proposal though